integration/restValidation.ts

```diff
@@ -0,0 +1,215 @@
+import {init as initBLS, SecretKey} from "@chainsafe/bls";
+import {LogLevel, WinstonLogger} from "@chainsafe/lodestar-utils";
+import {Validator} from "@chainsafe/lodestar-validator";
+import {config} from "@chainsafe/lodestar-config/lib/presets/mainnet";
+import {CgEth2ApiClient} from "../src/renderer/services/eth2/client/eth2ApiClient";
+import {CGSlashingProtection} from "../src/renderer/services/eth2/client/slashingProtection";
+import sinon from "sinon";
+import EventSource from "eventsource";
+import {computeEpochAtSlot} from "@chainsafe/lodestar-beacon-state-transition";
+import axios from "axios";
+
+const getValidatorIndexFromBits = (aggregationBits: string): number => {
+    const bitArr = config.types.CommitteeBits.fromJson(aggregationBits);
+    let firstIndex = -1;
+    bitArr.forEach((bool, index) => {
+        if (bool) {
+            // in case aggregate and proof we dont want to use index
+            if (firstIndex !== -1) return -1;
+            firstIndex = index;
+        }
+    });
+    return firstIndex;
+};
+
+type CommitteesRaw = {
+    index: string;
+    slot: string;
+    validators: string[];
+};
+type Committees = {
+    index: number;
+    slot: number;
+    validators: number[];
+};
+const getCommitteesFactory = (baseUrl: string) => async (
+    validatorIndex: number,
+    blockSlot: number,
+    ignoreBefore?: number,
+): Promise<Committees[]> => {
+    const url = new URL(`/eth/v1/beacon/states/${blockSlot}/committees`, baseUrl);
```

Comment:
can we use beacon api client rather than re-implementing everywhere?

API will receive changes over time and it will be nightmare to update on bunch of places.

Reply:
what if the validator beacon API master class has something wrongly implemented?

Reply:
we override if there is non compliant api: https://github.com/NodeFactoryIo/ChainGuardian/pull/541#discussion_r597486236